Packs/Base/Scripts/DBotTrainClustering/DBotTrainClustering.py

```diff
@@ -153,8 +150,10 @@ def reduce_dimension(self, dimension=2):
         :return:
         """
         if not self.TSNE_:
-            tsne = TSNE(n_jobs=-1, n_components=dimension, learning_rate=1000)
-            self.data_2d = tsne.fit_transform(pd.DataFrame(self.centers).T)
+            samples = pd.DataFrame(self.centers).T
+            perplexity = float(min(30, samples.shape[0] - 1))
+            tsne = TSNE(perplexity=perplexity, n_jobs=-1, n_components=dimension, learning_rate=1000)
```

Comment:
lets check out what this change entails, what is perplexity and what is happening/whats the impact by lowering it?